pkg/analytics/analytics.go

```diff
@@ -225,13 +226,17 @@ func (a *AnalyticsImpl) GetOutputData() *analyticsOutput {
 		}
 	}
 
-	// deepcode ignore InsecureHash: It is just being used to generate an id, without any security concerns
-	//nolint:gosec // sha1 only used to generate an id
-	shasum := sha1.New()
-	//nolint:errcheck // breaking api change needed to fix this
-	uuid, _ := uuid.GenerateUUID()
-	//nolint:errcheck // breaking api change needed to fix this
-	io.WriteString(shasum, uuid)
+	shasum := sha256.New() // generating a sha256 is not necessary, but gets rid of linter and sec warnings
+	generatedUUID, err := uuid.GenerateUUID()
+	if err != nil {
```

Comment:
Issue: this is a change of behaviour, in the previous version errors have been ignored, now they lead to returning nil. Similar changes have already created bugs, which is why I wouldn't just do it on the side.

Reply:
If you can't generate a UUID this needs to fail. Either change the return parameters to add the error, or panic. I'd prefer the first, but I'd accept the latter (a broken secure rng is probably a good reason to crash).